lib/crypto.js

```diff
@@ -75,21 +35,106 @@ function rsa(method, defaultPadding) {
   };
 }
 
-exports.publicEncrypt = rsa(binding.publicEncrypt,
-                            constants.RSA_PKCS1_OAEP_PADDING);
-exports.publicDecrypt = rsa(binding.publicDecrypt,
-                            constants.RSA_PKCS1_PADDING);
-exports.privateEncrypt = rsa(binding.privateEncrypt,
-                             constants.RSA_PKCS1_PADDING);
-exports.privateDecrypt = rsa(binding.privateDecrypt,
-                             constants.RSA_PKCS1_OAEP_PADDING);
-
-exports.pbkdf2 = function(password,
-                          salt,
-                          iterations,
-                          keylen,
-                          digest,
-                          callback) {
+module.exports = exports = {
+  createCipher: internalCipher.createCipher,
+  createCipheriv: internalCipher.createCipheriv,
+  createDecipher: internalCipher.createDecipher,
+  createDecipheriv: internalCipher.createDecipheriv,
+  createDiffieHellman: internalDh.createDiffieHellman,
+  createDiffieHellmanGroup: internalDh.createDiffieHellmanGroup,
+  createECDH: internalDh.createECDH,
+  createHash: internalHash.createHash,
+  createHmac: internalHash.createHmac,
+  createSign: internalSign.createSign,
+  createVerify: internalSign.createVerify,
+  getDiffieHellman: internalDh.createDiffieHellmanGroup,
+  getCiphers: cachedResult(() => filterDuplicateStrings(getCiphers())),
+  getCurves: cachedResult(() => filterDuplicateStrings(getCurves())),
+  getHashes: cachedResult(() => filterDuplicateStrings(getHashes())),
+  pbkdf2: pbkdf2,
+  pbkdf2Sync: pbkdf2Sync,
+  publicEncrypt: rsa(binding.publicEncrypt, constants.RSA_PKCS1_OAEP_PADDING),
+  publicDecrypt: rsa(binding.publicDecrypt, constants.RSA_PKCS1_PADDING),
+  privateEncrypt: rsa(binding.privateEncrypt, constants.RSA_PKCS1_PADDING),
+  privateDecrypt: rsa(binding.privateDecrypt, constants.RSA_PKCS1_OAEP_PADDING),
+  prng: randomBytes,
+  pseudoRandomBytes: randomBytes,
+  randomBytes: randomBytes,
+  rng: randomBytes,
+  setEngine: setEngine,
+  timingSafeEqual: timingSafeEqual,
+  Certificate: internalCert.createCertificate,
+  Cipher: internalCipher.createCipher,
+  Cipheriv: internalCipher.createCipheriv,
+  Decipher: internalCipher.createDecipher,
+  Decipheriv: internalCipher.createDecipheriv,
+  DiffieHellman: internalDh.createDiffieHellman,
+  DiffieHellmanGroup: internalDh.createDiffieHellmanGroup,
+  Hash: internalHash.createHash,
+  Hmac: internalHash.createHmac,
+  Sign: internalSign.createSign,
+  Verify: internalSign.createVerify,
+
+  _toBuf: toBuf
+};
+
+Object.defineProperties(exports, {
```

Comment:
Well, the point of doing the `module.exports = { ... }` is to avoid hidden classes, so these last few `Object.defineProperties()` are creating hidden classes AFAIK (akin to what was being done before -- `exports.foo = bar`).